### ISSUE TITLE
setCurrentItem on viewPager not working

### ISSUE THREAD

**Issue by IGZJcMinarro** [jpardogo/PagerSlidingTabStrip#58]
When I create my view, I want that viewPager move to some pager (eg pager number 1).
ViewPager move to that pager and PSTS update the states of every tab, but doesn't move the scroll to the correct position.
I have changed your example to be like mine and it doesn't work. 
The changes that I have done are the following:

MainActivity.java:
```java
@Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.inject(this);
        setSupportActionBar(toolbar);
        // create our manager instance after the content view is set
        mTintManager = new SystemBarTintManager(this);
        // enable status bar tint
        mTintManager.setStatusBarTintEnabled(true);
        adapter = new MyPagerAdapter(getSupportFragmentManager());
        pager.setAdapter(adapter);
        tabs.setViewPager(pager);
        final int pageMargin = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 4, getResources()
                .getDisplayMetrics());
        pager.setPageMargin(pageMargin);
        changeColor(getResources().getColor(R.color.green));
	 pager.setCurrentItem(1);
    }
```

activity_main.xml:
```xml
<com.astuetz.PagerSlidingTabStrip
        android:id="@+id/tabs"
        android:layout_width="match_parent"
        android:layout_height="?attr/actionBarSize"
        android:background="?attr/colorPrimary"
	    app:pstsShouldExpand="true" />
```

![screenshot_2015-01-24-13-26-51](https://cloud.githubusercontent.com/assets/4047514/5887383/0dd29d08-a3cd-11e4-97b1-3458119e3b89.png)

**Comment by jpardogo** [comment]
Change:
````java
     pager.setCurrentItem(1);
```
For:
```java
    pager.post(new Runnable() {
            @Override
            public void run() {
                pager.setCurrentItem(1);
            }
    });
```

When changing the default tab, there are still problems keeping the highlight and offset values so you may notice 'snapping' or double highlight tabs... 
I will try to spend some time when possible on it. because it has been a problem for a while and I probably didn't spend enough time looking for a proper solution. 
So far, that's all you can get changing the default starting app.
I leave the issue open.

**Comment by IGZJcMinarro** [comment]
Ok, that works, but is a bit rare...
It starts on position 0 and then scroll to position x...

**Comment by jpardogo** [comment]
Yes I know. I want that fixed. If you find the solution PR are welcome. Otherwise I will try to fix it when I have a moment to study the problem in depth

**Comment by IGZJcMinarro** [comment]
I have seen that if you don't remove the `OnGlobalLayoutListener` that you added on `notifyDataSetChanged()` method, It works fine. But I don't know if it could made other bugs.

**Comment by jpardogo** [comment]
Fix in PR #65 and example updated to demonstrate fix in commit 5672cc8076612a3642c577061f89b2be7122b73e

**Comment by szuwest** [comment]
Is there any way to not snapping?
